bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/handler/EpsonProjectorHandler.java

```diff
@@ -0,0 +1,410 @@
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EpsonProjectorHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Pauli Anttila, Yannick Schaus - Initial contribution
+ * @author Michael Lobstein - Improvements for OH3
+ */
+@NonNullByDefault
+public class EpsonProjectorHandler extends BaseThingHandler {
+    private static final int DEFAULT_POLLING_INTERVAL_SEC = 10;
+
+    private final Logger logger = LoggerFactory.getLogger(EpsonProjectorHandler.class);
+
+    private EpsonProjectorDevice device = new EpsonProjectorDevice();
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private final SerialPortManager serialPortManager;
+
+    private boolean isPowerOn = false;
+
+    public EpsonProjectorHandler(Thing thing, SerialPortManager serialPortManager) {
+        super(thing);
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getId();
+        if (command instanceof RefreshType) {
+            Channel channel = this.thing.getChannel(channelUID);
+            if (channel != null) {
+                updateChannelState(channel);
+            }
+        } else {
+            EpsonProjectorCommandType epsonCommand = EpsonProjectorCommandType.getCommandType(channelId);
+            sendDataToDevice(epsonCommand, command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        scheduler.execute(() -> {
+            EpsonProjectorConfiguration config = getConfigAs(EpsonProjectorConfiguration.class);
+            final String serialPort = config.serialPort;
+            final String host = config.host;
+            final Integer port = config.port;
+            final Integer pollingInterval = config.pollingInterval;
+
+            if (serialPort != null && !serialPort.equals("")) {
+                device = new EpsonProjectorDevice(serialPortManager, serialPort, scheduler);
+            } else if (host != null && !host.equals("") && port > 0 && port < 65535) {
+                device = new EpsonProjectorDevice(host, port, scheduler);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+                return;
+            }
```

Comment:
Configuration validation shouldn't be done asynchronously and instead should be done inside the `initialize()` method.

Reply:
done